compiler/luci/pass/src/FuseTransposeWithMeanPass.cpp

```diff
@@ -0,0 +1,137 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseTransposeWithMeanPass.h"
+
+#include <luci/IR/CircleNode.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+namespace
+{
+luci::CircleConst *create_fused_indices(luci::CircleConst *rindices,
+                                        const std::vector<uint32_t> &fused_rindices)
+{
+  assert(rindices != nullptr); // FIX_CALLER_UNLESS
+
+  if (rindices->dtype() != loco::DataType::S32)
+    return nullptr;
+
+  assert(fused_rindices.size() == rindices->size<loco::DataType::S32>());
+
+  auto fused_rindices_const = luci::clone(rindices);
+  auto name = rindices->name();
+  assert(name.length() > 0); // FIX_CALLER_UNLESS
+  fused_rindices_const->name(name + "_orig");
+
+  for (uint32_t i = 0; i < fused_rindices.size(); ++i)
+  {
+    fused_rindices_const->at<loco::DataType::S32>(i) = fused_rindices.at(i);
+  }
+
+  return fused_rindices_const;
+}
+
+bool const_has_value_s32(const luci::CircleConst *circle_const, int32_t value)
+{
+  if (circle_const->dtype() != loco::DataType::S32)
+    return false;
+
+  uint32_t size = circle_const->size<loco::DataType::S32>();
+  for (uint32_t i = 0; i < size; ++i)
+  {
+    if (circle_const->at<loco::DataType::S32>(i) == value)
+      return true;
+  }
+
+  return false;
+}
+
+bool fuse_transpose_with_mean(luci::CircleMean *mean)
+{
+  auto transpose = dynamic_cast<luci::CircleTranspose *>(mean->input());
+  if (not transpose)
+    return false;
+
+  // Get reduction indices of CircleMean operation.
+  auto rindices = dynamic_cast<luci::CircleConst *>(mean->reduction_indices());
+  if (not rindices)
+    return false;
+
+  assert(rindices->dtype() == loco::DataType::S32);
+
+  if (mean->keep_dims() != false)
+    return false;
+
+  auto perm = dynamic_cast<luci::CircleConst *>(transpose->perm());
+  if (not perm)
+    return false;
+
+  std::vector<uint32_t> axes_after_reduction;
+  std::vector<uint32_t> orig_reduced_axes;
+  for (uint32_t axis = 0; axis < perm->size<loco::DataType::S32>(); ++axis)
+  {
+    uint32_t original_axis = static_cast<uint32_t>(perm->at<loco::DataType::S32>(axis));
+
+    if (const_has_value_s32(rindices, axis))
+    {
+      orig_reduced_axes.push_back(original_axis);
+      continue;
+    }
+
+    axes_after_reduction.push_back(original_axis);
+  }
+
+  if (not std::is_sorted(axes_after_reduction.begin(), axes_after_reduction.end()))
+    return false;
+
+  // Create and configure new CircleMean operation.
+  auto fused_mean = mean->graph()->nodes()->create<luci::CircleMean>();
+  fused_mean->reduction_indices(create_fused_indices(rindices, orig_reduced_axes));
```

Comment:
`create_fused_indices` can return nullptr but I don't think `reduction_indices` with nullptr is OK.